src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala

```diff
@@ -857,6 +850,26 @@ trait AppAndGroupFormats {
         "secrets" -> runSpec.secrets,
         "taskKillGracePeriodSeconds" -> runSpec.taskKillGracePeriod
       )
+      // top-level ports fields are incompatible with IP/CT
+      if (runSpec.ipAddress.isEmpty) {
+        appJson = appJson ++ Json.obj(
+          // the ports field was written incorrectly in old code if a container was specified
+          // it should contain the service ports
+          //
+          // "ports" -> runSpec.servicePorts, -- can't specify both this and portDefinitions and be valid,
```

Comment:
Removing this will break mesosphere/marathon-lb (see https://github.com/mesosphere/marathon-lb/blob/d43d1fb9f8c9ed41af9becb5ef8d32897da0aa79/utils.py#L114) and probably other tools as well.